proxy/connect/connect.go

```diff
@@ -0,0 +1,186 @@
+package connect
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/v14/forwardrpc"
+	"github.com/stripe/veneur/v14/proxy/grpcstats"
+	"github.com/stripe/veneur/v14/samplers/metricpb"
+	"github.com/stripe/veneur/v14/scopedstatsd"
+	"google.golang.org/grpc"
+)
+
+type Connect interface {
+	Connect(context.Context, string, DestinationHash) (Destination, error)
+}
+
+var _ Connect = &connect{}
+
+type connect struct {
+	dialTimeout time.Duration
+	logger      *logrus.Entry
+	statsd      scopedstatsd.Client
+}
+
+func Create(
+	dialTimeout time.Duration, logger *logrus.Entry, statsd scopedstatsd.Client,
+) Connect {
+	return &connect{
+		dialTimeout: dialTimeout,
+		logger:      logger,
+		statsd:      statsd,
+	}
+}
+
+type Destination interface {
+	SendChannel() chan SendRequest
+	Close()
+}
+
+type SendRequest struct {
+	Metric       *metricpb.Metric
+	ErrorChannel chan error
```

Comment:
Because I want to scope the response to the request. I could use two channels, one for request and one for response, but in that case the reader and writer both have to be single threads; otherwise if two threads write to the request channel and concurrently read from the response channel, their responses could be interchanged. Channels are cheap, so the overhead of doing this is minimal.